### ISSUE TITLE
The EventEmitter class conflicts the EventEmitter module

### ISSUE THREAD

**Issue by koshigoe** [aws/aws-sdk-ruby#1791]
### Issue description

The [`event_emitter`](https://github.com/shokai/event_emitter) gem implements `EventEmitter` module.
The `aws-sdk-core` gem implements `EventEmitter` class.

- https://github.com/shokai/event_emitter/blob/a01fe7850bd48995bd424146263b6ddb45fd763e/lib/event_emitter/emitter.rb
- https://github.com/aws/aws-sdk-ruby/blob/821a66f491c33ecffdcbfcbfddbb8dcf9be2d630/gems/aws-sdk-core/lib/aws-sdk-core/event_emitter.rb

So, I can't use `event_emitter` together with `aws-sdk-core`.
(`TypeError: EventEmitter is not a module`)

### Gem name ('aws-sdk', 'aws-sdk-resources' or service gems like 'aws-sdk-s3') and its version

- aws-sdk-core 3.21.1

### Version of Ruby, OS environment

```
$ uname -v
Darwin Kernel Version 17.5.0: Fri Apr 13 19:32:32 PDT 2018; root:xnu-4570.51.2~1/RELEASE_X86_64
$ ruby -v
ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-darwin17]
```

### Code snippets / steps to reproduce

```
# Gemfile
source 'https://rubygems.org'

gem 'aws-sdk-core'
gem 'event_emitter'
```

```
$ bundle exec irb
(001): >> require 'aws-sdk-core'
=> true
(002): >> require 'event_emitter'
Traceback (most recent call last):
       16: from .../lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli.rb:18:in `start'
       15: from .../lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
       14: from .../lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli.rb:27:in `dispatch'
       13: from .../lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
       12: from .../lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
       11: from .../lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
       10: from .../lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli.rb:424:in `exec'
        9: from .../lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:28:in `run'
        8: from .../lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `kernel_load'
        7: from .../lib/ruby/gems/2.5.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `load'
        6: from .../bin/irb:11:in `<top (required)>'
        5: from (irb):2
        4: from (irb):2:in `require'
        3: from .../vendor/bundle/gems/event_emitter-0.2.6/lib/event_emitter.rb:4:in `<top (required)>'
        2: from .../vendor/bundle/gems/event_emitter-0.2.6/lib/event_emitter.rb:4:in `require'
        1: from .../vendor/bundle/gems/event_emitter-0.2.6/lib/event_emitter/emitter.rb:2:in `<top (required)>'
TypeError (EventEmitter is not a module)
```

**Comment by cjyclaire** [comment]
@koshigoe Fix PR #1792 is under review, I'll work on including this in the next release (likely tomorrow)
Feel free to try out the patch :) Thank you for your patience!

**Comment by cjyclaire** [comment]
Latest release [v 3.21.2](https://rubygems.org/gems/aws-sdk-core/versions/3.21.2) of `aws-sdk-core` is out, closing.
Feel free to reopen with further issue or questions :)
